src/main/java/io/cdap/plugin/aws/sqs/common/SqsPluginConfig.java

```diff
@@ -0,0 +1,263 @@
+  @Macro
+  private String queueName;
+
+  @Name(SqsConstants.PROPERTY_SQS_ENDPOINT)
+  @Description("The endpoint of the SQS server to connect to. For example, https://sqs.us-east-1.amazonaws.com")
+  @Nullable
+  @Macro
+  private String sqsEndpoint;
+
+  /**
+   * Constructor for SqsPluginConfig object.
+   * @param referenceName The reference name
+   * @param authenticationMethod The authentication method to be used
+   * @param accessId The AWS access id
+   * @param accessKey The AWS access key
+   * @param region The AWS region
+   * @param queueName The queue name
+   * @param sqsEndpoint The SQS Endpoint
+   */
+  public SqsPluginConfig(String referenceName, String authenticationMethod, @Nullable String accessId,
+                         @Nullable String accessKey, String region, String queueName, @Nullable String sqsEndpoint) {
+    super(referenceName);
+    this.authenticationMethod = authenticationMethod;
+    this.accessId = accessId;
+    this.accessKey = accessKey;
+    this.region = region;
+    this.queueName = queueName;
+    this.sqsEndpoint = sqsEndpoint;
+  }
+
+  /**
+   * Returns the auth method chosen.
+   *
+   * @param collector The failure collector to collect the errors
+   * @return An instance of SqsAuthMethod
+   */
+  public SqsAuthMethod getAuthenticationMethod(FailureCollector collector) {
+    SqsAuthMethod authMethod = getAuthenticationMethod();
+    if (authMethod != null) {
+      return authMethod;
+    }
+
+    collector.addFailure("Unsupported auth method value: " + authenticationMethod,
+                         String.format("Supported auth methods are: %s", SqsAuthMethod.getSupportedAuthMethods()))
+      .withConfigProperty(SqsConstants.PROPERTY_AUTH_METHOD);
+    collector.getOrThrowException();
+    return null;
+  }
+
+  /**
+   * Returns the auth method chosen.
+   *
+   * @return An instance of SqsAuthMethod
+   */
+  public SqsAuthMethod getAuthenticationMethod() {
+    Optional<SqsAuthMethod> sqsAuthMethod = SqsAuthMethod.fromValue(authenticationMethod);
+
+    return sqsAuthMethod.isPresent() ? sqsAuthMethod.get() : null;
+  }
+
+  @Nullable
+  public String getAccessId() {
+    return accessId;
+  }
+
+  @Nullable
+  public String getAccessKey() {
+    return accessKey;
+  }
+
+  /**
+   * Returns the region chosen.
+   *
+   * @param collector The failure collector to collect the errors
+   * @return An instance of Regions
+   */
+  public Regions getRegion(FailureCollector collector) {
+    Regions awsRegion = getRegion();
+    if (awsRegion != null) {
+      return awsRegion;
+    }
+
+    String supportedRegions = Arrays.stream(Regions.values()).map(Regions::getName)
+      .collect(Collectors.joining(", "));
+
+    collector.addFailure("Unsupported region value: " + region,
+      String.format("Supported regions are: %s", supportedRegions))
+      .withConfigProperty(SqsConstants.PROPERTY_REGION);
+
+    collector.getOrThrowException();
```

Comment:
`getOrThrowException()` will throw the exception if one exists. That is not recommended for methods that are called in the validation path. This `getRegion()` is called by `validateRegion()` so this should not throw an error. The collector should be used to collect all errors so they can be displayed all at once to the user.

This applies to the other get/validate methods in this class.

Reply:
Done